src/Parser/Core/Modules/StatTracker.js

```diff
@@ -0,0 +1,257 @@
+// TODO: stat constants somewhere else? they're largely copied from combatant
+class StatTracker extends Module {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  static STAT_BUFFS = {
+    //// POTIONS ////
+    [SPELLS.POTION_OF_PROLONGED_POWER.id]: { str: 2500, agi: 2500, int: 2500 },
+    // probably no need to add flasks/food because they never fall
+
+    //// TRINKETS ////
+    [SPELLS.SHADOWS_STRIKE.id]: {
+      itemId: ITEMS.DREADSTONE_OF_ENDLESS_SHADOWS.id,
+      crit: (_, item) => calculateSecondaryStatDefault(845, 3480, item.itemLevel),
+    },
+    [SPELLS.SHADOW_MASTER.id]: {
+      itemId: ITEMS.DREADSTONE_OF_ENDLESS_SHADOWS.id,
+      mastery: (_, item) => calculateSecondaryStatDefault(845, 3480, item.itemLevel),
+    },
+    [SPELLS.SWARMING_SHADOWS.id]: {
+      itemId: ITEMS.DREADSTONE_OF_ENDLESS_SHADOWS.id,
+      haste: (_, item) => calculateSecondaryStatDefault(845, 3480, item.itemLevel),
+    },
+
+    //// MISC ////
+    [SPELLS.JACINS_RUSE.id]: { mastery: 3000 },
+
+    //// DRUID ////
+    [SPELLS.ASTRAL_HARMONY.id]: { mastery: 4000 },
+
+
+  }
+
+  _startingStats = {};
+  _stats = {};
+
+  on_initialized() {
+    this._startingStats = {
+      str: this.combatants.selected.strength,
+      agi: this.combatants.selected.agility,
+      int: this.combatants.selected.intellect,
+      crit: this.combatants.selected.critRating,
+      haste: this.combatants.selected.hasteRating,
+      mastery: this.combatants.selected.masteryRating,
+      vers: this.combatants.selected.versatilityRating,
+    };
+    this._stats = this._startingStats;
+
+    debug && this._debugPrintStats(this._stats);
+  }
+
+  /*
+   * Stat rating at pull.
+   * Should be identical to what you get from Combatant.
+   */
+  get startingStrengthRating() {
+    return this._startingStats.str;
+  }
+  get startingAgilityRating() {
+    return this._startingStats.agi;
+  }
+  get startingIntellectRating() {
+    return this._startingStats.int;
+  }
+  get startingCritRating() {
+    return this._startingStats.crit;
+  }
+  get startingHasteRating() {
+    return this._startingStats.haste;
+  }
+  get startingMasteryRating() {
+    return this._startingStats.mastery;
+  }
+  get startingVersRating() {
+    return this._startingStats.vers;
+  }
+
+  /*
+   * Current stat rating, as tracked by this module.
+   */
+  get currentStrengthRating() {
+    return this._stats.str;
+  }
+  get currentAgilityRating() {
+    return this._stats.agi;
+  }
+  get currentIntellectRating() {
+    return this._stats.int;
+  }
+  get currentCritRating() {
+    return this._stats.crit;
+  }
+  get currentHasteRating() {
+    return this._stats.haste;
+  }
+  get currentMasteryRating() {
+    return this._stats.mastery;
+  }
+  get currentVersRating() {
+    return this._stats.vers;
+  }
+
+  /*
+   * For percentage stats, the percentage you'd have with zero rating.
+   * These values don't change.
+   */
+  get baseCritPercentage() {
+    return 0.08;
```

Comment:
Isn't this different for each class/spec?

Reply:
Uhh is it? It's currently implemented in Combatant as a flat 0.08.

Reply:
I guess we can hope it's right 🙏